File: docs/EnterpriseArchitect/Unidade.php
<?php
require_once ('Jovem.php');
require_once ('Empresa.php');
require_once ('Curso_Unidade.php');

namespace Configuracoes;



use Aprendizes;
use Empresas;
use Cursos;
/**
 * @author Cristiano
 * @version 1.0
 * @created 14-nov-2017 17:51:01
 */
class Unidade extends Curso_Unidade
{

	private $bairro;
	private $cEP;
	private $cidade;
	private $cNPJ;
	private $complemento;
	private $dataDeCadastro;
	private $email;
	private $endereco;
	private $estado;
	private $fax;
	private $nomeFantazia;
	private $numero;
	private $razaoSocial;
	private $status;
	private $telefone;
	private $unidade_id;

	function __construct()
	{
	}

	function __destruct()
	{
	}



	/**
	 * 
	 * @param unidade
	 */
	public function cadastrarUnidade(Unidade $unidade)
	{
	}

	/**
	 * 
	 * @param unidade
	 * @param unidade_id
	 */
	public function editarUnidade(Unidade $unidade, long $unidade_id)
	{
	}

	public function listarUnidades()
	{
	}

	/**
	 * 
	 * @param unidade_id
	 */
	public function visualizarUnidade(long $unidade_id)
	{
	}

}
?>

File: software/workspace/ipp/src/main/java/br/net/ipp/models/configuracoes/Unidade.java
package br.net.ipp.models.configuracoes;

import javax.persistence.Column;
import javax.persistence.Entity;

import org.hibernate.validator.constraints.NotEmpty;

import br.net.ipp.enums.Status;
import br.net.ipp.models.AbstractEntity;

@Entity
public class Unidade extends AbstractEntity {

	private static final long serialVersionUID = 1L;
	
	@NotEmpty
	private String nomeFantazia;
	@NotEmpty
	@Column(unique = true)
	private String email;
	private String razaoSocial;
	private Status status;
	private String cNPJ;
	private String fax;
	private String telefone;
	private String dataDeCadastro;
	private String endereco;
	private String numero;
	private String complemento;
	private String bairro;
	private String cidade;
	private String estado;
	private String cEP;
	
	public String getNomeFantazia() {
		return nomeFantazia;
	}
	public void setNomeFantazia(String nomeFantazia) {
		this.nomeFantazia = nomeFantazia;
	}
	public String getEmail() {
		return email;
	}
	public void setEmail(String email) {
		this.email = email;
	}
	public String getRazaoSocial() {
		return razaoSocial;
	}
	public void setRazaoSocial(String razaoSocial) {
		this.razaoSocial = razaoSocial;
	}
	public Status getStatus() {
		return status;
	}
	public void setStatus(Status status) {
		this.status = status;
	}
	public String getcNPJ() {
		return cNPJ;
	}
	public void setcNPJ(String cNPJ) {
		this.cNPJ = cNPJ;
	}
	public String getFax() {
		return fax;
	}
	public void setFax(String fax) {
		this.fax = fax;
	}
	public String getTelefone() {
		return telefone;
	}
	public void setTelefone(String telefone) {
		this.telefone = telefone;
	}
	public String getDataDeCadastro() {
		return dataDeCadastro;
	}
	public void setDataDeCadastro(String dataDeCadastro) {
		this.dataDeCadastro = dataDeCadastro;
	}
	public String getEndereco() {
		return endereco;
	}
	public void setEndereco(String endereco) {
		this.endereco = endereco;
	}
	public String getNumero() {
		return numero;
	}
	public void setNumero(String numero) {
		this.numero = numero;
	}
	public String getComplemento() {
		return complemento;
	}
	public void setComplemento(String complemento) {
		this.complemento = complemento;
	}
	public String getBairro() {
		return bairro;
	}
	public void setBairro(String bairro) {
		this.bairro = bairro;
	}
	public String getCidade() {
		return cidade;
	}
	public void setCidade(String cidade) {
		this.cidade = cidade;
	}
	public String getEstado() {
		return estado;
	}
	public void setEstado(String estado) {
		this.estado = estado;
	}
	public String getcEP() {
		return cEP;
	}
	public void setcEP(String cEP) {
		this.cEP = cEP;
	}
	public static long getSerialversionuid() {
		return serialVersionUID;
	}
	
}